### ISSUE TITLE
Configuration serialization should use a single representation.

### ISSUE THREAD

**Issue by theSloopJohnB** [NISystemsEngineering/CEF#92]
The serialization scheme right now uses a mix of json and xml. The json is used to represent the hierarchy of the nodes, but xml is used to represent the data within the nodes.

This scheme makes reading the configuration file painful.

**Comment by tannerblair** [comment]
I've been working on a DOM-based JSON library with Milan Raj that is a lot more flexible than the native solution. If you decide to move the config to JSON, I would recommend checking it out. It makes it much easier to manage JSON that may change.

https://github.com/rajsite/i3-json

**Comment by theSloopJohnB** [comment]
Thanks for the tip. I see that the tools network hasn't seen a release in over a year with any of your recent commits. Are you or Milan planning on releasing an updated VIPB soon?

**Comment by tannerblair** [comment]
Yeah, there are a few other features I was hoping to implement, namely better array handling on parse and some bug squashing with the error handling. I've gotten sucked in to a few other things, but if you think you might want to use it I can bump it up the priority queue.

**Comment by theSloopJohnB** [comment]
I’ll let you know if we decide to implement this feature in the short term. If we did start work on it in the short term, I could begin using the library while you’re still working on improving the internals of it, so long as we coordinate publishing to VIPM.

**Comment by smithed** [comment]
What would be the advantage of the i3 json library as compared to any of these? 
https://bitbucket.org/lavag/json-api-labview/downloads/
https://bitbucket.org/drjdpowell/jsontext
https://github.com/JKISoftware/JKI-JSON-Serialization
A quick glance seems to indicate that the i3 version is almost identical in principle to the lava version, drjdpowell's was built to overcome some performance limitations in that style, and the jki version seems to be "flatten to json, but done by people who use it"

With regards to the broader issue, fixing this would also resolve #80 as they are related (that issue focuses on making the module data be more sane and standard, rather than the scheme I came up with for performance.

**Comment by tannerblair** [comment]
I personally prefer i3 because it's wrapped better, is less verbose, and the wire in i3 is a variant so it's much easier to troubleshoot when there's a problem. The variant is also much easier to extend when it doesn't quite do what you want, and it allows for the ability to query a parsed structure to see what keys are present rather than having to know the name of all keys in advance.

There are also some nice functions for visualizing the data in a JSON tree, so you can go way beyond prettyprint when wanting to look at what's in a JSON Object created with this:

![image](https://user-images.githubusercontent.com/12645089/28480973-9af64d1a-6e28-11e7-8d55-35d5c306ec6d.png)


It's also written and maintained in LV 2014, so it's compatible with every LV version that DCAF supports.
jsontext is written in 2017 and uses VIMs, so can't be saved for previous.

It also has 0 dependencies, whereas JKI and LAVA both have dependencies on 4 or so OpenG libraries.

Also, all of our other JSON-based applications in SE are moving towards this so from a maintenance standpoint it would be nice to not splinter what JSON API we're using.

**Comment by smithed** [comment]
None of those seem like particularly great points for building yet another json library, but ok. You should read through this article http://seriot.ch/parsing_json.php and try not to fail as hard as the other json parsers of the world. 

With the tree functions I don't know if you used this as a baseline, but it might be good to compare notes with https://lavag.org/files/file/19-variant-probe/

**Comment by tannerblair** [comment]
To be clear, I didn't make this and I'm not making this. Milan Raj did. It's existed for 5 or so years now, I'm just helping to document and extend it.
